app/services/resource_service.rb

```diff
@@ -14,16 +14,32 @@ def create(resource_class, params)
     @resource_class = resource_class
     @resource = resource_class.new(params)
     @resource.data_provider = data_provider
-
-    # skip create if already exists
+    # skip create if record already exists and new record has the same attribute values as the new record and the record was not provided by maz
     external_resource = find_external_resource
-    return external_resource.external if external_resource.present?
+    old_record = external_resource.try(:external)
+    return old_record if external_resource.present? && unchanged_attributes?(old_record) && not_maz?(data_provider)
 
-    # create resource
-    create_external_resource if resource.save
+    # destroy the old record and its external_reference if its attributes changed or if the data_provider is MAZ
+    old_record.destroy if old_record.present?
+    # necessary because dependant: :destroy sometimes works for external resource and sometimes doesn't
+    # TODO: Remove line and make dependant: :destroy work
+    external_resource.destroy if external_resource.present?
+    create_external_resource if @resource.save
     resource_or_error_message(resource)
   end
 
+  def unchanged_attributes?(record)
+    return false if record.blank?
+
+    @resource.attributes.except("id", "created_at", "updated_at", "tag_list", "category_id", "region_id") == record.attributes.except("id", "created_at", "updated_at", "tag_list", "category_id", "region_id")
+  end
+
```

Comment:
```
def unchanged_attributes?(old_record)
  return false if old_record.blank?
  except_attributes = ["id", "created_at", "updated_at", "tag_list", "category_id", "region_id"]
  @resource.attributes.except(*except_attributes) == old_record.attributes.except(*except_attributes)
end
```

Reply:
bei einem NewsItem würden hier noch bsp folgende Attribute verglichen werden:

```
{"author"=>"Julienne Ferry", "full_version"=>true, "characters_to_be_shown"=>0, "publication_date"=>nil, "published_at"=>Sat, 19 May 0029 00:00:00 UTC +00:00, "show_publish_date"=>true, "news_type"=>"News article or story", "data_provider_id"=>40, "external_id"=>nil, "title"=>"Foo"}
```

Die `content_blocks`, `address` und `source_url` würden dabei ignoriert werden

Reply:
bei bsp einem POI würden dann folgende Attribute vergleichen werden:

```
{"name"=>"Burg 9", "description"=>"FooBAr", "mobile_description"=>"FooBar", "active"=>true, "length_km"=>nil, "means_of_transportation"=>nil, "type"=>"PointOfInterest", "data_provider_id"=>10}
```

Änderungen an den restlichen nested Attributes (und deren nested Attributes) führend dann zu keinem Update: 
```
certificates, regions, addresses, contact, media_contents, accessibility_information, operating_company, web_urls
```

Reply:
Ja das ist nicht wirklich befriedigend aber besser als nichts. Ansonsten können wir uns das die destroy und recreate action hier sparen und komplexe Logiken für tatsächliche update actions überlegen. Das war mir auf jeden Fall eine Lehre. In Zukunft werde ich auf nested_Attributes verzichten.